pkg/controller/performanceprofile/performanceprofile_controller.go

```diff
@@ -103,54 +210,350 @@ func (r *ReconcilePerformanceProfile) Reconcile(request reconcile.Request) (reco
+			return nil, err
+		}
+
+		updated, err := r.pauseMachineConfigPool(mcpUpdate, false)
+		if err != nil {
+			return nil, err
+		}
+
+		if updated {
+			return &reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+		}
+		return nil, nil
+	}
+
+	// create machine config pool and pause it
+	if mcpMutated != nil && created {
+		if err := r.createOrUpdateMachineConfigPool(mcpMutated); err != nil {
+			return nil, err
+		}
+	}
+
+	mcpUpdate, err := r.getMachineConfigPool(mcp.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	updated, err := r.pauseMachineConfigPool(mcpUpdate, true)
+	if err != nil {
+		return nil, err
+	}
+
+	// we want to give time to the machine-config controllers to get updated values
+	if updated {
+		return &reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+	}
+
+	// update the machine config pool after that we paused it
+	if mcpMutated != nil && !created {
+		if err := r.createOrUpdateMachineConfigPool(mcpMutated); err != nil {
+			return nil, err
+		}
+	}
+
+	if mcMutated != nil {
+		if err := r.createOrUpdateMachineConfig(mcMutated); err != nil {
+			return nil, err
+		}
+	}
+
+	if networkLatencyTunedMutated != nil {
+		if err := r.createOrUpdateTuned(networkLatencyTunedMutated); err != nil {
+			return nil, err
+		}
+	}
+
+	if realTimeKernelTunedMutated != nil {
+		if err := r.createOrUpdateTuned(realTimeKernelTunedMutated); err != nil {
+			return nil, err
+		}
+	}
+
+	if fgMutated != nil {
+		if err := r.createOrUpdateFeatureGate(fgMutated); err != nil {
+			return nil, err
+		}
+
+		// feature gate resource should be updated before KubeletConfig creation, otherwise
+		// we will lack TopologyManager feature gate under the kubelet configuration
+		// see - https://bugzilla.redhat.com/show_bug.cgi?id=1788061#c3
+		// we want to give time to the kubelet-config controllers to get updated feature gate resource
+		return &reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+	}
+
+	if kcMutated != nil {
+		if err := r.createOrUpdateKubeletConfig(kcMutated); err != nil {
+			return nil, err
+		}
+	}
+
+	return &reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+}
+
+func (r *ReconcilePerformanceProfile) pauseMachineConfigPool(mcp *mcov1.MachineConfigPool, pause bool) (bool, error) {
+	if mcp.Spec.Paused == pause {
+		return false, nil
+	}
+
+	mcp.Spec.Paused = pause
+	return true, r.client.Update(context.TODO(), mcp)
+}
+
+func (r *ReconcilePerformanceProfile) validatePerformanceProfileParameters(performanceProfile *performancev1alpha1.PerformanceProfile) error {
+	if performanceProfile.Spec.CPU == nil {
+		return fmt.Errorf("you should provide CPU section")
+	}
+
+	if performanceProfile.Spec.CPU.Isolated == nil {
+		return fmt.Errorf("you should provide isolated CPU set")
+	}
+
+	if performanceProfile.Spec.CPU.NonIsolated == nil {
+		return fmt.Errorf("you should provide non isolated CPU set")
+	}
+
+	if performanceProfile.Spec.RealTimeKernel == nil {
```

Comment:
@cynepco3hahue @davidvossel heads up, I am already working on a follow up PR for making this a boolean, because with OCP 4.4 we don't need the repo url anymore at all, the RPMs are in the RHCOS image already. Also I remove validation for this and just enable the RT kernel systemd unit when the boolean is set and true.

Reply:
@davidvossel  According to Marc comment, can we leave it for following PR, because it will need change under scripts and API?

Reply:
ack, @slintes thanks for following up on this.